### PR TITLE
[moe] support hybrid parallel

### PR DESCRIPTION
**Pull request by oahzxl**
support hybrid parallel with pp + dp + ep
* add custom shard policy for openmoe
* add moe hybrid plugin which modifies some init settings of original hybrid plguin
* updata parallel process group mesh in moe
* use layernorm kernel and jit fuse in shardformer

### CODE REVIEW

File: colossalai/booster/plugin/moe_hybrid_parallel_plugin.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+from colossalai.pipeline.stage_manager import PipelineStageManager
+from colossalai.shardformer import ShardConfig
+from colossalai.shardformer.policies.base_policy import Policy
+
+from .pp_plugin_base import PipelinePluginBase
+
+PP_AXIS, DP_AXIS, TP_AXIS = 0, 1, 2
+
+
+class MoeHybridParallelPlugin(PipelinePluginBase):
```
Comment (ver217) [inline colossalai/booster/plugin/moe_hybrid_parallel_plugin.py:33]: What's the difference between this plugin and hybrid parallel plugin?

Comment (oahzxl) [reply colossalai/booster/plugin/moe_hybrid_parallel_plugin.py:33]: their process grouph axis is different

Comment (ver217) [reply colossalai/booster/plugin/moe_hybrid_parallel_plugin.py:33]: Then can it inherit HybridParallelPlugin and overwrite a few methods like `__init__()`?

File: examples/language/openmoe/test_ci.sh [hunk 2]
```diff
@@ -1,5 +0,0 @@
-set -xe
-pip install -r requirements.txt
-
-python infer.py --model "test"
-torchrun --standalone --nproc_per_node 2 train.py --model_name "test" --batch_size 1 --num_epoch 20
```
Comment (ver217) [inline examples/language/openmoe/test_ci.sh:5]: When can you run ci?

File: colossalai/booster/plugin/moe_hybrid_parallel_plugin.py (hunk 1 above)
Comment (oahzxl) [reply colossalai/booster/plugin/moe_hybrid_parallel_plugin.py:33]: ok

File: examples/language/openmoe/test_ci.sh (hunk 2 above)
Comment (oahzxl) [reply examples/language/openmoe/test_ci.sh:5]: ci fails due to some kernel initialization. will be updated soon

**Review by CWHer — APPROVED** [review]

**Pull request merged**